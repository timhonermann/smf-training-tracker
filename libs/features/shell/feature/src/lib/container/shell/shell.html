<stt-header />
<div class="content">
  <router-outlet />
</div>
<stt-menu [menuItems]="menuItems()" />
