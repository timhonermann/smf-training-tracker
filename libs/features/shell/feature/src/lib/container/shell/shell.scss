:host {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;

  .content {
    flex: 1;
    padding: 10px;
    overflow: hidden;
  }
}
