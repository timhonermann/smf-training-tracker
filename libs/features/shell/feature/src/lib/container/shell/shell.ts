import { ChangeDetectionStrategy, Component, signal } from '@angular/core';
import { Header, Menu } from '@stt/features/shell/ui';
import { RouterOutlet } from '@angular/router';
import { featureRoutes } from '@stt/shared/routing/model';
import { MenuItemData } from '@stt/features/shell/model';

@Component({
  selector: 'stt-shell',
  imports: [Header, RouterOutlet, Menu],
  templateUrl: './shell.html',
  styleUrl: './shell.scss',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class Shell {
  readonly menuItems = signal<MenuItemData[]>([
    {
      path: featureRoutes.DASHBOARD,
      label: 'Dashboard',
      icon: 'dashboard',
    },
    {
      path: featureRoutes.TRAINING,
      label: 'Training',
      icon: 'paw',
    },
    {
      path: featureRoutes.PERSON,
      label: 'People',
      icon: 'person',
    },
  ]).asReadonly();
}
