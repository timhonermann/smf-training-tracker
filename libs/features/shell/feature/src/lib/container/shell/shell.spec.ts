import { ComponentFixture, TestBed } from '@angular/core/testing';
import { Shell } from './shell';
import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
import { expect } from 'vitest';

describe('Shell', () => {
  let component: Shell;
  let fixture: ComponentFixture<Shell>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [Shell],
    })
      .overrideComponent(Shell, {
        set: { imports: [], schemas: [CUSTOM_ELEMENTS_SCHEMA] },
      })
      .compileComponents();

    fixture = TestBed.createComponent(Shell);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('menuItems', () => {
    expect(component.menuItems()).toMatchSnapshot();
  });
});
