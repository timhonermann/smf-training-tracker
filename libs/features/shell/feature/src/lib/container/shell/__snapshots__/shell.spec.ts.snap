// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shell > menuItems 1`] = `
[
  {
    "icon": "dashboard",
    "label": "Dashboard",
    "path": "dashboard",
  },
  {
    "icon": "paw",
    "label": "Training",
    "path": "training",
  },
  {
    "icon": "person",
    "label": "People",
    "path": "person",
  },
]
`;
